gtsfm/utils/verification.py

```diff
@@ -74,38 +79,65 @@ def essential_to_fundamental_matrix(
     return np.linalg.inv(camera_intrinsics_i2.K().T) @ i2Ei1.matrix() @ np.linalg.inv(camera_intrinsics_i1.K())
 
 
-def compute_epipolar_distances(
-    normalized_coords_i1: np.ndarray, normalized_coords_i2: np.ndarray, i2Ei1: EssentialMatrix
+def compute_epipolar_distances_sed(
+    coordinates_i1: np.ndarray, coordinates_i2: np.ndarray, i2Fi1: np.ndarray
 ) -> Optional[np.ndarray]:
-    """Compute symmetric point-line epipolar distances between normalized coordinates of correspondences.
+    """Compute symmetric point-line epipolar distance between corresponding coordinates in two images. The SED is the
+    geometric point-line distance and is a biased estimate of the gold-standard reprojection error.
+
+    References: 
+    - "Fathy et al., Fundamental Matrix Estimation: A Study of Error Criteria"
 
     Args:
-        normalized_coords_i1: normalized coordinates in image i1, of shape Nx2.
-        normalized_coords_i2: corr. normalized coordinates in image i2, of shape Nx2.
-        i2Ei1: essential matrix between two images
+        coordinates_i1: coordinates in image i1, of shape Nx2.
+        coordinates_i2: corr. coordinates in image i2, of shape Nx2.
+        i2Fi1: fundamental matrix between two images.
 
     Returns:
-        Symmetric epipolar distances for each row of the input, of shape N.
+        Symmetric epipolar point-line distances for each row of the input, of shape N.
     """
-    if (
-        normalized_coords_i1 is None
-        or normalized_coords_i1.size == 0
-        or normalized_coords_i2 is None
-        or normalized_coords_i2.size == 0
-    ):
+
+    if coordinates_i1 is None or coordinates_i1.size == 0 or coordinates_i2 is None or coordinates_i2.size == 0:
+        return None
+
+    epipolar_lines_i2 = feature_utils.convert_to_epipolar_lines(coordinates_i1, i2Fi1)  # Ex1
+    epipolar_lines_i1 = feature_utils.convert_to_epipolar_lines(coordinates_i2, i2Fi1.T)  # Etx2
+
+    numerator = feature_utils.point_line_dotproduct(coordinates_i1, epipolar_lines_i1)
+
+    line_sq_norms_i1 = np.sum(np.square(epipolar_lines_i1[:, :2]), axis=1)
+    line_sq_norms_i2 = np.sum(np.square(epipolar_lines_i2[:, :2]), axis=1)
+
+    return numerator * np.sqrt(1 / line_sq_norms_i1 + 1 / line_sq_norms_i2)
```

Comment:
need to resolve if Equaiton 13 is equivlalent https://openaccess.thecvf.com/content_ECCV_2018/papers/Rene_Ranftl_Deep_Fundamental_Matrix_ECCV_2018_paper.pdf

Reply:
check his code

Reference: HZ page 288